proxy_agent/src/main.rs

```diff
@@ -72,22 +86,47 @@ fn main() {
     } else {
         #[cfg(windows)]
         {
-            _ = service_dispatcher::start(constants::PROXY_AGENT_SERVICE_NAME, ffi_service_main);
+            //
+            let main = std::thread::spawn(|| {
+                _ = service_dispatcher::start(
+                    constants::PROXY_AGENT_SERVICE_NAME,
+                    ffi_service_main,
+                );
+            });
+
+            main.join().unwrap();
         }
 
         #[cfg(not(windows))]
         {
-            service::start_service_wait();
+            service::start_service_wait().await;
         }
     }
 }
 
 #[cfg(windows)]
-fn proxy_agent_windows_service_main(args: Vec<OsString>) {
+fn proxy_agent_windows_service_main(_args: Vec<OsString>) {
     // start the Instant to calculate the elapsed time
     _ = helpers::get_elapsed_time_in_millisec();
 
-    if let Err(e) = windows::run_service(args) {
-        logger::write_error(format!("{e}"));
+    match ASYNC_RUNTIME_HANDLE.get() {
+        Some(handle) => {
+            handle.spawn(windows::run_service());
+        }
+        None => {
+            logger::write_error("Failed to get the tokio runtime handle.".to_string());
+        }
```

Comment:
I don't think it makes sense for the service to continue running in this case. This is a fatal programmer error since this can only happen if `ASYNC_RUNTIME_HANDLE` wasn't initialized prior to this function being invoked. Something like `let handle = ASYNC_RUNTIME_HANDLE.get().expect("You must provide the Tokio runtime handle before this function is called")` or something to that effect.